### PR TITLE
Add yarn support

### PR DESCRIPTION
**Pull request by maxtuzz**

### CODE REVIEW

**Review by fredcooke — APPROVED** [review]
Again, new functionality best to be out there getting use/testing. Patch later if need be.

**Comment by stickycode** [comment]
Woaa there, some tests maybe and i have some feed back

File: branchout-yarn [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+function yarnSettings() {
+    if test -z "${BRANCHOUT_NPM_REPOSITORY}"; then
+    readValue "your maven repository" "BRANCHOUT_NPM_REPOSITORY"
+    [[ ${BRANCHOUT_NPM_REPOSITORY} =~ ^https:// ]] || BRANCHOUT_NPM_REPOSITORY="https://${BRANCHOUT_NPM_REPOSITORY}"
+    "${BRANCHOUT_PATH}/branchout" set "BRANCHOUT_NPM_REPOSITORY" "${BRANCHOUT_NPM_REPOSITORY}"
+    fi
+    
+    readValue "npm repository username" "NPM_USER"
```
Comment (stickycode) [inline branchout-yarn:18]: Use ensureValue that will check Branchoutfile first and put it back

File: branchout-yarn [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    readSecret "npm repository secret" "NPM_PASS"
+    readValue "npm repository email (used for identification):" "NPM_EMAIL"
```
Comment (stickycode) [inline branchout-yarn:20]: Use ensureConfigValue, that will save/use value from branchout home/branchoutrc.

I think the email will be the same as used in the commit author email so saving is a good idea

Comment (maxtuzz) [reply branchout-yarn]: Good call! Updated and tested. Works well.

**Comment by stickycode** [comment]
@maxtuzz @fredcooke Added fairly complete coverage

* settings generation
* reused of existing settings
* saving of settings from interactive generation
* adding show command that lets you see the yarnrc

**Pull request merged**